openpype/modules/sync_server/sync_server.py

```diff
@@ -396,6 +398,8 @@ async def sync_loop(self):
                                               representation,
                                               site,
                                               error)
+                    # Trigger files process finished
+                    self.files_are_processed = False
```

Comment:
I think this should be moved a tab left, here you are in loop for single project.

I am about bit confused about 'files_are_processed' set to False here, I would expect they would be set to True. (Logic seems to be in opposite direction just because of check of response.content in rest api).

And I don't think it will work as expected if you manage to trigger download of workfile if there is already loop that is synchronizing in progress. The loop will mark that is `files_are_processed` but these are completely different files, your workfile won't be being downloaded yet.

I think safer would be loop and check progress or finished synch for particular representation.

Reply:
Do you think this check should be requested through REST API and executed by the server or only on the client side?

Reply:
I think only on client side. 
Eventually it will be changed to some kind of REST api call to v4 server, but I wouldn't worry about it in this PR.

Reply:
Do you have any clue about where to listen for such information on the client side?

Reply:
I added a `priority` field to the `add_site` function because `update_db` wasn't working and I had no clue why. This works pretty well.